templates/problem/problem.html

```diff
@@ -322,6 +331,40 @@ <h2 style="display: inline-block">{{ title }}</h2>
             </div>
         </div>
     {% endif %}
+
+    {% set contest = contest_problem.contest %}
+    {% if contest_problems %}
+    <div id="contest-problems-list">
+        <div class="toggle open unselectable">
+            <i class="fa fa-chevron-right fa-fw"></i>{{ _('Problems') }}
+        </div>
+        <div class="toggled">
+            <table class="table striped">
+                <tbody>
+                    {% for contest_problem in contest_problems %}
+                        <tr>
+                            {% if problem.code == contest_problem.code %}
+                            <td class="highlighted" style="text-align: center">
+                            {% else %}
+                            <td style="text-align: center">
+                            {% endif %}
+
+                            {% if can_view_all_problems or problem.is_public %}
+                                <a href="{{ url('problem_detail', contest_problem.code) }}">
+                                    {{contest.get_label_for_problem(loop.index0)}}
+                                </a>
+                            {% else %}
+                                {{contest.get_label_for_problem(loop.index0)}}
```

Comment:
for in-contest problems, they are not public, so the check will always be false, aka the link is not show -> useless